app/lib/authorized_only.rb

```diff
@@ -0,0 +1,58 @@
+# Dispatches different questions about authorization
+# to specific methods for specific models.
+class AuthorizedOnly
+  def initialize(authorizer)
+    @authorizer = authorizer
+  end
+
+  # Branches by type (Relation, Array, Model)
+  # and by the Model class.
+  def dispatch(&block)
+    return_value = block.call
+
+    # Can't duck type here, since other things like hashes
+    # are iterable or mappable in ways we don't want to
+    # support.
+    if return_value.class < ActiveRecord::Relation # subclass?
+      filter_array(return_value.to_a)
+    elsif return_value.class == Array # subclass?
```

Comment:
These look swapped to me. It calls `filter_relation` for array and `filter_array` for relation.

Reply:
thanks!